crates/topos-p2p/src/runtime/handle_event.rs

```diff
@@ -67,19 +71,64 @@ impl
 
                 self.active_listeners.insert(listener_id);
             }
+
+            SwarmEvent::OutgoingConnectionError {
+                connection_id,
+                peer_id: Some(peer_id),
+                error,
+            } if self
+                .state_machine
+                .successfully_connect_to_bootpeer
+                .is_none()
+                && self.state_machine.dialed_bootpeer.contains(&connection_id) =>
+            {
+                warn!("Unable to connect to bootpeer {peer_id}: {error:?}");
+                self.state_machine.dialed_bootpeer.remove(&connection_id);
+                if self.state_machine.dialed_bootpeer.is_empty() {
+                    // We tried to connect to all bootnode without success
+                    error!("Unable to connect to any bootnode");
+                }
```

Comment:
Shouldn't we retry (with exponential backoff) here?

Reply:
To expand on that, the goal is to make it so that the TCE can get up and running regardless of the order in which nodes start/stop, and to recover from various adverse network conditions. So what if the bootnodes are busy restarting?

Reply:
This PR is to allow nodes to start in any order. The recovering or any other behaviour are not handled yet.

The retries are handled not manually but rely on the frequency of the `bootstrap` queries.
The dialling of the bootnode is triggered by the initialization of a bootstrap query.
The `dialed_bootpeer` is handling single dialing for one bootnode as represented by the `ConnectionId`.